### DISCUSSION TITLE
How to allow a C++ function exposed by pybind to take in a reference of C++ class object exposed by nanobind?

### DISCUSSION THREAD

**Discussion by aashaka** [wjakob/nanobind#355] [General]
I have exposed the C++ class `SimpleClass` to Python using nanobind and after setting some parameters, I would like to call a C++ function `setup_from_simple_class` which is exposed to Python using pybind. I see a type mismatch: the C++ side expects `test::SimpleClass` arg and receives `test._test.SimpleClass`. (Here test is my namespace).

I cannot change which library is used for bindings, so my question is it possible to do type conversions etc. in order to allow pybinded C++ function to take in a nanobinded object class?

**Comment by wjakob** [comment]
There is no transparent inter-op between pybind11 and nanobind because this would force nanobind to inherit all of the complexities of pybind11. The reason of existence of this project was to get rid of this complexity, so this is not a solvable issue.

Nanobind can treat a pybind11 object like any other Python object (``nb::object``) and access fields or call methods etc. that way (read up on *wrapper types* in the docs for details). If you want a lower C++-level interop, you will have to redo the bindings of both tools with the same binding toolkit.
